// Copyright 2018 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// See platform2/media_perception/mojom/pipeline.mojom for relevant
// comments on data fields.

syntax = "proto2";

package mri;

enum PipelineStatus {
  PIPELINE_STATUS_UNKNOWN = 0;
  STARTED = 1;
  RUNNING = 2;
  SUSPENDED = 3;
  ERROR = 4;
}

enum PipelineErrorType {
  PIPELINE_ERROR_TYPE_UNKNOWN = 0;
  CONFIGURATION = 1;
  STARTUP = 2;
  RUNTIME = 3;
  CONTENT = 4;
}

message PipelineError {
  optional PipelineErrorType error_type = 1;
  optional string error_source = 2;
  optional string error_string = 3;
}

message PipelineState {
  optional PipelineStatus status = 1;
  optional PipelineError error = 2;
  optional string configuration_name = 3;
}

message GlobalPipelineState {
  repeated PipelineState states = 1;
}

